server/src/main/java/org/apache/druid/server/coordinator/DataSourceCompactionConfig.java

```diff
@@ -104,6 +100,43 @@ public DataSourceCompactionConfig(
     );
   }
 
+  /**
+   * Copied from CompactionTask#getValidTargetCompactionSizeBytes.
+   * Must be synced.
+   */
```

Comment:
May be not in this PR but better to create a common method for both ?

Reply:
Hmm, they are not exactly same because `numShards` is missing in `DataSourceCompactionConfig`. I'm not sure how to do it well.

Rather, I think we need to merge `server` and `indexing-service` modules somehow, so that the coordinator and the overlord use the same classes for ingestion spec. The biggest problem here is the dependency on `druid-indexing-hadoop` which isn't necessary for all modules depending on `server`.